src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+
+const MALE = 'm';
+const FEMALE = 'f';
+
+interface Props {
+  people: Person[];
+  delay: number;
+  onSelected?: (user: Person) => void,
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [hasFocus, setHasFocus] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+    applyQuery(event.target.value);
+    setHasFocus(true);
+    if (!event.target.value) {
+      setHasFocus(false);
+    }
```

Comment:
You use `event.target.value` three times so it can be put in a separate variable.